### DISCUSSION TITLE
Randomly appearing input value results duplicated headers in output file (force blank column output)

### DISCUSSION THREAD

**Discussion by shungabubus** [johnkerl/miller#566] [Q&A] [answered]
Hello,

I was unable to find anything similar with key search so please help.

Because my source data has a parameter not always present after the pivot creation despite my efforts I don't have a fixed structure. Can I somehow enforce to keep the structure and place substitute character where Ledger value is missing?

**Command:**
mlr --icsvlite --ocsv reshape -i Process,Encoding,Ledger -o Process,Encoding,Ledger -s Setup,Value work.txt >test.txt

**Input data** (notice that ledger appears only at some lines the rest are fixed):
Company,Setup,Value
103,Process,no
103,Encoding,UTF-8
210,Process,yes
210,Encoding,UTF-8
210,Ledger,SA
305,Process,yes
305,Encoding,UTF-8
310,Process,yes
310,Encoding,UTF-8
311,Process,yes
311,Encoding,UTF-8
312,Process,yes
312,Encoding,UTF-8
315,Process,yes
315,Encoding,UTF-8
317,Process,yes
317,Encoding,UTF-8
317,Ledger,SA

**Output result:**
Company,Process,Encoding
103,no,UTF-8

Company,Process,Encoding,Ledger
210,yes,UTF-8,SA

Company,Process,Encoding
305,yes,UTF-8
310,yes,UTF-8
311,yes,UTF-8
312,yes,UTF-8
315,yes,UTF-8

Company,Process,Encoding,Ledger
317,yes,UTF-8,SA

Expected Output result:
Company,Process,Encoding,Ledger
103,no,UTF-8,
210,yes,UTF-8,SA
305,yes,UTF-8,
310,yes,UTF-8,
311,yes,UTF-8,
312,yes,UTF-8,
315,yes,UTF-8,
317,yes,UTF-8,SA

Many thanks:
Tamas

**Comment by aborruso** [comment, marked as answer]
@shungabubus it should be this one

```
mlr --csvlite reshape -s Setup,Value then unsparsify input.csv
```

You don't have to use `-i` because you must do long to wide reshape. Then you must use `unsparsify` to normalize field names absent in a given record but present in others.

**Reply by shungabubus** [reply]
Brilliant thank you so much. Works like a charm.
